app/core/pokemon-entity.ts

```diff
@@ -591,11 +591,45 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
   }
 
   addItem(item: Item, permanent = false) {
+    const type = SynergyGivenByItem[item]
+    if (type && this.types.has(type)) {
+      return // prevent adding synergy items for existing types
+    }
+
     this.items.add(item)
     this.simulation.applyItemEffect(this, item)
     if (permanent && !this.isGhostOpponent) {
       this.refToBoardPokemon.items.add(item)
     }
+
+    const addSynergyFromItem = () => {
+      this.types.add(type)
+      if (type === Synergy.GHOST) {
```

Comment:
We need to find a way to reuse `simulation.applySynergyEffects` here, in order to avoid duplicating all the logic. How about a second `synergy` parameter that filters the SynergyEffects applied ?

Reply:
I refactored applySynergyEffects for this and it's so much better